apps/desktop/src/lib/forge/github/githubPrMonitor.ts

```diff
@@ -56,15 +57,29 @@ export class GitHubPrMonitor implements ForgePrMonitor {
 		}
 	}
 
-	// Right after pushing GitHub will respond with status 422, necessatating retries.
+	/**
+	 * Get the PR information.
+	 *
+	 * Right after pushing GitHub will respond with status 422, necessatating retries.
+	 * After that, the mergeable state is not always available immediately.
+	 * The function will set the initial PR information immediately and then poll for the mergeable state.
+	 */
 	private async getPrWithRetries(prNumber: number): Promise<DetailedPullRequest> {
 		const request = async () => await this.prService.get(prNumber);
 		let lastError: any;
 		let attempt = 0;
-		while (attempt < 3) {
+		while (attempt < MAX_POLL_ATTEMPTS) {
 			attempt++;
 			try {
-				return await request();
+				const detailedPR = await request();
+				// Set the PR info immediately to show
+				// the user all preliminary information.
+				this.pr.set(detailedPR);
+				if (detailedPR.mergeableState === 'unknown') {
+					await sleep(1000);
+					continue;
+				}
+				return detailedPR;
```

Comment:
@mtsgrd, what I did here was that as soon as we have the detailed PR information, we set to the writable.
The function does not return yet, an continues polling as long as the `mergeableState` is `unknown` or the max poll attempts are crossed.
Only then the function returns and the `loading` writable is set to false.

That way we can display PR detailed info as soon as we get it, while also ensuring that the `mergeableStatus` will be fetched.

Reply:
Perfect.